flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DeclarativeSlotManager.java

```diff
@@ -200,10 +200,12 @@ public void suspend() {
 		LOG.info("Suspending the slot manager.");
 
 		resourceTracker.clear();
-		taskExecutorManager.close();
+		if (taskExecutorManager != null) {
```

Comment:
Maybe add `@Nullable` to `taskExecutorManager`.

Reply:
Actually we should maybe also add `@Nullable` to `resourceManagerId` and `resourceActions`.

Reply:
IIRC you suggested the same thing during the original review; my argument back then was that it would create a lot of unnecessary warnings.
I'm not sure what the better option is; but I suppose having any warning, even if they are maybe unnecessary, is better than none?

Reply:
I guess having warnings could have made us aware of the problem in suspend.